core/src/main/scala/cats/data/NonEmptySet.scala

```diff
@@ -23,22 +23,54 @@ import cats.{Always, Eq, Eval, Foldable, Later, Now, Reducible, SemigroupK, Show
 
 import scala.collection.immutable._
 
-final class NonEmptySet[A] private (val set: SortedSet[A]) {
+trait Newtype { self =>
+  private[data] type Base
+  private[data] trait Tag extends Any
+  private[cats] type Type[A] <: Base with Tag
+}
+
+private[data] object NonEmptySetImpl extends NonEmptySetInstances with Newtype {
```

Comment:
shall we override the toString as well?

Reply:
We can't really do that, since we can't actually add methods on the type, but only through implicit extension methods. This means that it will always call  the `toString` of the underlying type, we can of course use `show` though.